PWGMM/Tasks/analyse-mft-tracks.cxx

```diff
@@ -0,0 +1,114 @@
+
+struct analyseMFTTracks {
+  int icoll = 0;
+
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"TracksPhiEta_in_coll", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},
+      {"TracksPhiEta", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},                                                           //
+      {"TracksPhiEtaGen", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, 0, 2 * M_PI}, {35, -4.5, -1.}}}},                                                        //
+      {"TracksEtaZvtx", "; #eta; Z_{vtx}; tracks", {HistType::kTH2F, {{35, -4.5, -1.}, {201, -20.1, 20.1}}}},                                                          //
+      {"NtrkZvtx", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}},                                                         //
+      {"NtrkEta", "; N_{trk}; #eta; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {35, -4.5, -1.}}}},                                                                //
+      {"Multiplicity", "alibi_nightlies/O2DPG_pp_minbias_testbeam.sh/15-11-2021-18:00 - tf13; collisionID; N_{trk}^{MFT}", {HistType::kTH1F, {{101, 0., 100.}}}, true} //
+    }                                                                                                                                                                  //
+  };
+
+  void process(o2::aod::Collision const& collision, o2::aod::MFTTracks const& tracks)
+  {
+
+    auto z = collision.posZ();
+    registry.fill(HIST("NtrkZvtx"), tracks.size(), z);
+
+    for (auto& track : tracks) {
+      registry.fill(HIST("TracksPhiEta_in_coll"), track.phi(), track.eta());
+      registry.fill(HIST("TracksEtaZvtx"), track.eta(), z);
+      registry.fill(HIST("Multiplicity"), icoll);
+      registry.fill(HIST("NtrkEta"), tracks.size(), track.eta());
+    }
+    icoll++;
+  }
+  //end of process
+
+  //aod::McCollisions
+};
+//end of MyTask
+
+struct analyseGenTracks {
+
+  HistogramRegistry registryGen{
+    "registryGen",
+    {
+      {"TracksPhiEtaGen", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, 0, 2 * M_PI}, {35, -4.5, -1.}}}},   //
+      {"TracksEtaZvtxGen", "; #eta; Z_{vtx}; tracks", {HistType::kTH2F, {{35, -4.5, -1.}, {201, -20.1, 20.1}}}},  //
+      {"NtrkZvtxGen", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}}, //
+      {"NtrkEtaGen", "; N_{trk}; #eta; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {35, -4.5, -1.}}}},        //
+    }                                                                                                             //
+  };
+
+  void process(aod::McCollisions::iterator const& mcCollision, Particles const& particles)
+  {
+    int nChargedPrimaryParticles = 0;
+    auto z = mcCollision.posZ();
+
+    for (auto& particle : particles) {
+      auto p = TDatabasePDG::Instance()->GetParticle(particle.pdgCode());
```

Comment:
Please refer to how it is implemented in dNdeta task, you can declare `Service<TDatabasePDG> pdg;` inside the task. In fact, you do not need the code below, you can safely skip the particles that have no database entry (`p == nullptr`).

Reply:
If I do like you, I have the seg fault I told you about

Reply:
My code does not produce a segfault, so probably you are not using it correctly. Did you add the `Service<>` declaration in your task?

Reply:
Well if I run your code with my latest AO2D I have a segfault. But my code runs fine if I don't put this line of code, so I would really prefer to keep it as is. And I added the Service<> declaration yes :)

Reply:
Still, please use the same code, then we can investigate the source of segfault. It could be caused by input data, or maybe you were testing during the short period where O2 dev was broken (last two days).

Reply:
Ok I will do that then